torch/_dynamo/variables/user_defined.py

```diff
@@ -441,6 +464,36 @@ def call_function(
                 args[0],
                 mutation_type=ValueMutationNew(),
             )
+        elif edict and self.value is edict:
+            ed = edict()
+            if args:
+                assert len(args) == 1
+                assert isinstance(args[0], variables.ConstDictVariable)
+                ed.update(
+                    **{
+                        k.vt.as_python_constant(): v.as_python_constant()
+                        for k, v in args[0].items.items()
+                    }
+                )
+            if kwargs:
+                ed.update(**{k: v.as_python_constant() for k, v in kwargs.items()})
+
+            def build_key_value(k, v):
+                key = variables.ConstantVariable.create(k)
+                source_key = k
+
+                source_value = GetItemSource(self.source, source_key)
+                value = variables.LazyVariableTracker.create(v, source_value)
+
+                return key, value
+
+            items = dict(build_key_value(k, v) for k, v in ed.items())
+            dict_vt = variables.ConstDictVariable(
+                items, mutation_type=ValueMutationNew()
+            )
+            return UserDefinedDictVariable(
+                ed, dict_vt=dict_vt, mutation_type=AttributeMutationNew()
+            )
```

Comment:
@StrongerXi Sorry for the bother. Could you please take a look at here? As `edict` allows both value mutation and attribute mutation, which mutation type should be set on UserDefinedDictVariable?

Reply:
Aha good call, but I think the attribute mutation logic will be indexed on `UserDefinedDictVariable` and the value mutation on its `_dict_vt` field, which is a `ConstDictVariable`.

With that being said, I'd suggest look into why `edict` doesn't work out of the box. If you search for `(dict)` inside `test/dynamo/test_dicts.py`, you'll see lots of examples that just inherit from `dict` and work: 
https://github.com/pytorch/pytorch/blob/bb987492302e18ed5a8324a6e18d99553947367b/test/dynamo/test_dicts.py#L183-L199